File: spoj/PRIME/2.c
#include <stdio.h>
#include <malloc.h>
#include <math.h>

int is_prime(int digit){
   int up;
   int i;
   if (digit < 2){
       return 0;
   }
   if (digit < 4){
       return 1;
   }
   up = sqrt(digit) + 1;
   for(i=2; i < up; i++){
       if (digit % i == 0){
           return 0;
       }
    }
    return 1;
}

void print_prime(int start, int end){
    int i;
    for(i=start; i<end+1; i++){
        if (is_prime(i)){
            printf("%d\n", i);
        }
    }
}

int main(){
    int num;
    int start, end;
    int i;
    scanf("%d", &num);
    for(i=0; i<num; i++){
        scanf("%d %d", &start, &end);
        print_prime(start, end);
        if (i != num-1){
            printf("\n");
        }
    }
    return 0;
}

File: spoj/PRIME/2.py
#coding: utf-8

def get_primes(start, end):
    """Get the primes between start and end
    """
    out = []
    for i in range(start, end+1):
        if is_prime(i):
            out.append(i)
    return out


def is_prime(digit):
    """judge whether digit is a prime or not
    """
    if digit < 2:
        return False
    if digit < 4:
        return True
    #import math
    #up = int(math.sqrt(digit))
    up = int(digit ** 0.5)
    for i in range(2, up+1):
        if digit % i == 0:
            return False
    return True

if __name__ == "__main__":
    tests_num = int(raw_input())
    out = []
    for i in range(tests_num):
        line = raw_input()
        start, end = line.strip().split(" ")
        start = int(start)
        end = int(end)
        primes = get_primes(start, end)
        out.append(primes)
    i = 0
    for x in out:
        for a in x:
            print a
        if i != tests_num-1:
            print
        i += 1



